### PR TITLE
Fixed issue of duplicate message on whishlist

### LINKED ISSUE #26898: Alert Message On Product Page
When adding a product from wishlist to cart, alert messages are displaying more than once.
The number of alert messages is equal to the number of custom options.
image

### Preconditions (*)

1. Add products with custom options to the wishlist.
2. Add the product from wishlist to cart.

### Steps to reproduce (*)

1. Add the product from wishlist to cart.

### Expected result (*)

1. A single line of Alert message should get displayed.

### Actual result (*)

1. Alert messages are equal to the number of custom options. ( More than one alert message is displayed.)

### PR DESCRIPTION
**Pull request by deep701**
### Description (*)
When adding a product from wishlist to cart, alert messages are displaying more than once.
The number of alert messages is equal to the number of custom options.

1) Alert messages will be shown for individual products.
2) Have resolved the issue of number of message equal to the number of custom options. So now only single error message will be shown for particular product even after product has multiple custom options.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#26898: Alert Message On Product Page

### Manual testing scenarios (*)
1. Add products with required custom options to the wishlist.
2. Add the product from wishlist to cart.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: app/code/Magento/Catalog/Model/Product/Type/AbstractType.php [hunk 1]
```diff
@@ -9,9 +9,11 @@
 use Magento\Catalog\Api\ProductRepositoryInterface;
 use Magento\Framework\App\Filesystem\DirectoryList;
 use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\Filesystem\DriverInterface;
 
 /**
- * @api
```
Comment (rogyar) [inline app/code/Magento/Catalog/Model/Product/Type/AbstractType.php:14]: Please, do not remove the `@api` directive. It's an important marker for backward compatibility checks.

File: app/code/Magento/Catalog/Model/Product/Type/AbstractType.php [hunk 2]
```diff
@@ -492,7 +496,7 @@ public function processFileQueue()
                         /** @var $uploader \Zend_File_Transfer_Adapter_Http */
                         $uploader = isset($queueOptions['uploader']) ? $queueOptions['uploader'] : null;
 
-                        $path = dirname($dst);
+                        $path = DriverInterface::getParentDirectory($dst);
```
Comment (rogyar) [inline app/code/Magento/Catalog/Model/Product/Type/AbstractType.php]: Please, do not use static calls for non-static methods.

File: app/code/Magento/Catalog/Model/Product/Type/AbstractType.php [hunk 3]
```diff
@@ -507,13 +511,9 @@ public function processFileQueue()
...
                         $uploader->setDestination($path);
 
-                        if (empty($src) || empty($dst) || !$uploader->receive($src)) {
-                            /**
-                             * @todo: show invalid option
-                             */
-                            if (isset($queueOptions['option'])) {
-                                $queueOptions['option']->setIsValid(false);
-                            }
+                        if ((empty($src) || empty($dst) || !$uploader->receive($src)) && isset($queueOptions['option'])) {
```
Comment (rogyar) [inline app/code/Magento/Catalog/Model/Product/Type/AbstractType.php]: Please, make sure the line length does not exceed 120 characters.

**Review by rogyar — CHANGES_REQUESTED** [review]
Hi @deep701. Thank you for your collaboration. Please, check my comments below for minor improvements. Also, according to the [definition of done](https://devdocs.magento.com/guides/v2.3/contributor-guide/contributing_dod.html) all changes should be covered using automated tests. I would kindly ask you to cover your fix using MFTF test. 

Thank you!

**Comment by engcom-Echo** [comment]
I will take care of test coverage.

**Comment by engcom-Echo** [comment]
These changes badly affect the ability to create an order with the file option in the Admin panel.

image

**Comment by engcom-Echo** [comment]
Hi @deep701, I'm closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for your collaboration.

**Pull request closed without merging**